arangod/Cluster/TraverserEngine.cpp

```diff
@@ -187,30 +185,31 @@ void BaseEngine::getVertexData(VPackSlice vertex, VPackBuilder& builder,
       // Maybe handle differently
     }
 
-    if (shouldProduceVertices) {
-      std::string_view vertex = id.substr(pos + 1);
-      for (std::string const& shard : shards->second) {
-        Result res = _trx->documentFastPathLocal(
-            shard, vertex, [&](LocalDocumentId const&, VPackSlice doc) {
-              // FOUND. short circuit.
-              read++;
-              builder.add(v);
-              if (!options().getVertexProjections().empty()) {
-                VPackObjectBuilder guard(&builder);
-                options().getVertexProjections().toVelocyPackFromDocument(
-                    builder, doc, _trx.get());
-              } else {
-                builder.add(doc);
-              }
-              return true;
-            });
-        if (res.ok()) {
-          break;
-        }
-        if (res.isNot(TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND)) {
-          // We are in a very bad condition here...
-          THROW_ARANGO_EXCEPTION(res);
-        }
+    if (!shouldProduceVertices) {
+      return;
+    }
+    std::string_view vertex = id.substr(pos + 1);
+    for (std::string const& shard : shards->second) {
+      Result res = _trx->documentFastPathLocal(
+          shard, vertex, [&](LocalDocumentId const&, VPackSlice doc) {
+            // FOUND. short circuit.
+            read++;
+            builder.add(v);
+            if (!options().getVertexProjections().empty()) {
+              VPackObjectBuilder guard(&builder);
+              options().getVertexProjections().toVelocyPackFromDocument(
+                  builder, doc, _trx.get());
+            } else {
+              builder.add(doc);
+            }
+            return true;
+          });
+      if (res.ok()) {
+        break;
+      }
+      if (res.isNot(TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND)) {
+        // We are in a very bad condition here...
+        THROW_ARANGO_EXCEPTION(res);
```

Comment:
okay, but required? :D

Reply:
Not required. It is a matter of taste, I think.
I prefer early exit and as few indentation levels as possible. IMO that makes code more readable. But I don't claim that this is the only valid perspective.